### PR TITLE
Upgrade Ansible collection to the newest version

### PR DESCRIPTION
**Pull request by filip-szweda**
# Changes
## Requirements
* `Dynatrace-OneAgent-Ansible` now requires Ansible >= 2.15.0 (declared in readme, requirements.txt, meta/runtime.yml)
* pywinrm >= 0.4.1 is only required on Windows
## Porting
* explicitly declared unique filter as case insensitive ([source](https://docs.ansible.com/ansible/latest/porting_guides/porting_guide_core_2.12.html#plugins))
* added bool filter wherever needed ([source](https://docs.ansible.com/ansible/latest/porting_guides/porting_guide_core_2.12.html))
  * **Note to Reviewers: Should I filter through bool all variables, which should be booleans?**
## Fixes
* placed `ansible.builtin.include_tasks` outside other tasks (previous approach resulted in errors)
* updated how lists are defined (previous approach resulted in errors)
* removed `>` from `variable: >` (previous approach caused unnecessary `\n` to appear)
## Readme
* added `Requirements` section with `General` and `Windows` subsections
* added `Setup` section
* added `Build` section
* extended `Installation` section
## .gitignore
* built collections are now ignored
# Playbooks used in testing
## Installation with downloaded installer
```
---
- name: >
    Download and install
  hosts: localhost
  connection: local 
  collections:
    - dynatrace.oneagent
  vars:
    oneagent_package_state: present
    ansible_become: true
    oneagent_download_dir: <download_dir>
    oneagent_environment_url: <environment_url>
    oneagent_paas_token: <token>
    
  tasks:
    - name: Import Dynatrace OneAgent role
      ansible.builtin.import_role:
        name: oneagent
```
## Installation
```
---
- name: >
    Install
  hosts: localhost
  connection: local 
  collections:
    - dynatrace.oneagent
  vars:
    oneagent_package_state: present
    ansible_become: true
    oneagent_download_dir: /home/labuser
    oneagent_local_installer: /home/labuser/Dynatrace-OneAgent-Ansible/Dynatrace-OneAgent-Linux-<version>.sh
    
  tasks:
    - name: Import Dynatrace OneAgent role
      ansible.builtin.import_role:
        name: oneagent

```
## `oneagentctl` configuration
```
---
- name: >
    Configure
  hosts: localhost
  connection: local 
  collections:
    - dynatrace.oneagent
  vars:
    oneagent_package_state: present
    ansible_become: true
    oneagent_install_args:
      - --set-host-name=new_host_name
      - --set-host-tag=new_tag
      - --set-host-property=property1
      - --set-host-property=other=property
    
  tasks:
    - name: Import Dynatrace OneAgent role
      ansible.builtin.import_role:
        name: oneagent
```
## Uninstallation
```
---
- name: >
    Uninstall
  hosts: localhost
  connection: local 
  collections:
    - dynatrace.oneagent
  vars:
    oneagent_package_state: absent
    ansible_become: true
    
  tasks:
    - name: Import Dynatrace OneAgent role
      ansible.builtin.import_role:
        name: oneagent
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,29 @@
 # Dynatrace OneAgent collection
-
-In its current state, collection consist of a single role that deploys Dynatrace OneAgent on Linux and Windows operating systems using dedicated configuration and ensures the OneAgent service maintains a running state.
+In its current state, the collection consists of a single role that deploys Dynatrace OneAgent on Linux and Windows operating systems using dedicated configuration and ensures the OneAgent service maintains a running state.
 
 ## Requirements
+* Ansible >= 2.15.0
+* pywinrm >= 0.4.1
```
Comment (filip-szweda) [inline README.md:6]: `pywinrm` is only required on Windows.

File: README.md [hunk 2]
```diff
@@ -1,24 +1,29 @@
...
+## Setup
+`pip install -r requirements.txt`
 
-* Ansible >= 2.9.0
-* pywinrm >= 0.4.1
+## Build
+`ansible-galaxy collection build .`
 
 ## Installation
+To install the latest stable release of the collection on your system, call:
```
[the remark this answers is not in the document]
Comment (filip-szweda) [reply README.md:15]: Documentation Team is not in charge of READMEs

File: roles/oneagent/tasks/provide-installer/download-unix.yml [hunk 3]
```diff
@@ -10,7 +10,7 @@
   ansible.builtin.get_url:
     url: "{{ _oneagent_installer_url }}"
     dest: "{{ oneagent_installer_path }}"
-    validate_certs: "{{ oneagent_validate_certs | bool }}"
```
[the remark this answers is not in the document]
Comment (filip-szweda) [reply roles/oneagent/tasks/provide-installer/download-unix.yml:13]: I haven't found a place in the codebase where it is set to a string. In fact, from what I see, it is always set to `true`. Am I missing something?
https://github.com/Dynatrace/Dynatrace-OneAgent-Ansible/blob/7957517678cddd007207ba50e3f49718e8d1933a/roles/oneagent/defaults/main.yml#L32

**Comment by WStechura** [comment]
> `Applying OneAgent configuration` step does not seem to be working properly

File: roles/oneagent/README.md [hunk 4]
```diff
@@ -41,7 +41,7 @@ The following variables are available in `defaults/main/` and can be overridden:
 
 | Name | Default | Description
 |-|-|-
-| `oneagent_environment_url` | `-` | The URL of the target Dynatrace environment (SaaS or Managed).
+| `oneagent_environment_url` | `-` | The URL of the target Dynatrace environment (SaaS or Managed), see [Direct download from your environment](#direct-download-from-your-environment) for format.
```
Comment (WStechura) [inline roles/oneagent/README.md:44]: 1. The URL of the target Dynatrace environment (see [Direct download from your environment](#direct-download-from-your-environment))

**Pull request merged**